website/docs/r/cos_replication.html.markdown

```diff
@@ -0,0 +1,171 @@
+  service           = "cloud-object-storage"
+  plan              = "standard"
+  location          = "global"
+}
+
+resource "ibm_resource_instance" "cos_destination_instance" {
+  name              = "cos-instance-dest"
+  resource_group_id = data.ibm_resource_group.cos_group.id
+  service           = "cloud-object-storage"
+  plan              = "standard"
+  location          = "global"
+}
+
+resource "ibm_cos_bucket" "source_bucket" {
+  bucket_name           = "a-bucket-source"
+  resource_instance_id = ibm_resource_instance.cos_source_instance.id
+  region_location      = "us-south"
+  storage_class         = "standard"
+  object_versioning {
+    enable  = true
+  }
+}
+
+resource "ibm_cos_bucket" "destination_bucket" {
+  bucket_name           = "a-bucket-destination"
+  resource_instance_id = ibm_resource_instance.cos_destination_instance.id
+  region_location      = "us-south"
+  storage_class         = "standard"
+  object_versioning {
+    enable  = true
+  }
+}
+
+
+### Configure IAM authorization policy
+
+resource "ibm_iam_authorization_policy" "policy" {
+  roles                  = [
+      "Writer",
+  ]
+  subject_attributes {
+    name  = "accountId"
+    value = "an-account-id"
+  }
+  subject_attributes {
+    name  = "serviceName"
+    value = "cloud-object-storage"
+  }
+  subject_attributes {
+    name  = "serviceInstance"
+    value = ibm_resource_instance.cos_source_instance.guid
+  }
+  subject_attributes {
+    name  = "resource"
+    value = ibm_cos_bucket.source_bucket.bucket_name
+  }
+  subject_attributes {
+    name  = "resourceType"
+    value = "bucket"
+  }
+  resource_attributes {
+    name     = "accountId"
+    value    = "an-account-id"
+  }
+  resource_attributes {
+    name     = "serviceName"
+    value    = "cloud-object-storage"
+  }
+  resource_attributes { 
+    name  =  "serviceInstance"
+    value =  ibm_resource_instance.cos_destination_instance.guid
+  }
+  resource_attributes { 
+    name  =  "resource"
+    value =   ibm_cos_bucket.destination_bucket.bucket_name
+  }
+  resource_attributes { 
+    name  =  "resourceType"
+    value =  "bucket" 
+  }
+}
+
+### Configure replication policy
+
+resource "ibm_cos_bucket_replication_rule" "cos_bucket_repl" {
+  depends_on = [
+      ibm_iam_authorization_policy.policy
+  ]
+  bucket_crn	    = ibm_cos_bucket.source_bucket.crn
+	bucket_location = ibm_cos_bucket.source_bucket.region_location
+  replication_rule {
+    rule_id = "a-rule-id"
+    enable = "true"
+    prefix = "a-prefix"
+    priority = "a-priority-associated-with-the-rule"
+    deletemarker_replication_status = "Enabled/Suspened"
+    destination_bucket_crn = ibm_cos_bucket.destination_bucket.crn
+  }
+}
+
+```
+
+## Argument reference
+Review the argument references that you can specify for your resource. 
+- `bucket_crn` - (Required, Forces new resource, String) The CRN of the COS bucket.
+- `bucket_location` - (Required, Forces new resource, String) The location of the COS bucket.
```

Comment:
endpoint is optional

and replication_rule is required so it should come into Arguemnt section not in Attribute section

Reply:
done